src/octokit/get-octokit-throttle-options.ts

```diff
@@ -10,7 +16,27 @@ type Options = {
 export function getOctokitThrottleOptions(options: Options) {
   let { log, redisConfig } = options;
 
-  if (!redisConfig) return;
+  if (!redisConfig)
+    return {
+      onRateLimit: (retryAfter, options: { [key: string]: any }) => {
+        log.warn(
+          `Request quota exhausted for request ${options.method} ${options.url}`
+        );
+
+        // Retry twice after hitting a rate limit error, then give up
+        if (options.request.retryCount <= 2) {
+          log.info(`Retrying after ${retryAfter} seconds!`);
+          return true;
+        }
+        return false;
+      },
+      onSecondaryRateLimit: (retryAfter, options: { [key: string]: any }) => {
+        // does not retry, only logs a warning
+        log.warn(
+          `Secondary quota detected for request ${options.method} ${options.url}`
+        );
+      },
+    } as ThrottlingOptions;
```

Comment:
can we store the throttle base options as a variable to avoid code duplication with the code below?

Reply:
done